source/solution_deploy/source/action_target_provider.py

```diff
@@ -0,0 +1,152 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+"""Custom resource provider for AWS Security Hub Custom Action Target"""
+
+# Test Event
+# {
+#     "ResourceProperties": {
+#         "Name": "Remediate with ASR",
+#         "Description": "Submit the finding to Automated Security Response on AWS",
+#         "Id": "ASRRemediation"
+#     },
+#     "RequestType": "Create",
+#     "ResponseURL": "https://bogus"
+# }
+
+import os
+import json
+import boto3
+from botocore.config import Config
+from botocore.exceptions import ClientError
+from logger import Logger
+import cfnresponse
+
+# initialize logger
+LOG_LEVEL = os.getenv("log_level", "info")
+logger_obj = Logger(loglevel=LOG_LEVEL)
+REGION = os.getenv("AWS_REGION", "us-east-1")
+PARTITION = os.getenv("AWS_PARTITION", default="aws")  # Set by deployment template
+
+BOTO_CONFIG = Config(retries={"mode": "standard"})
+CLIENTS = {}
+
+
+def get_securityhub_client():
+    if "securityhub" not in CLIENTS:
+        CLIENTS["securityhub"] = boto3.client("securityhub", config=BOTO_CONFIG)
+    return CLIENTS["securityhub"]
+
+
+class InvalidCustomAction(Exception):
+    pass
+
+
+class CustomAction(object):
+    """
+    Security Hub CustomAction class
+    """
+
+    name = ""
+    description = ""
+    id = ""
+    account = ""
+
+    def __init__(self, account, properties):
```

Comment:
This class is lacking coherence. We have to initialize the class by passing accountId, name, description and id.
But accountId is only used in `delete` and the other 3 parameters are only used in `create`.

So instead of putting all that into the constructor, we should make accountId a parameter of `delete` and the other 3 parameters of `create`. (As a side effect, this makes the unit tests for each method shorter.)

Then it becomes obvious that `create` and `delete` do not share state - they don't want to be a class at all, they want to be pure standalone functions.

Reply:
I agree. It's not new code. Due to the complexity of custom resources, I'm not inclined to change anything here without thorough testing of the upgrade workflow. This is a change to make early in a release cycle.